source/constructs/api/search/main.py

```diff
@@ -118,3 +126,36 @@ def tables():
     for searchable_class in searchable:
         tables.append(searchable_class.__tablename__)
     return tables
+
+
+@router.get("/download-logs", response_class=StreamingResponse)
+def download_log_as_zip():
+    filename = f"aws_sdps_cloudwatch_logs.zip"
+    logs = boto3.client('logs')
+    response = logs.describe_log_groups(logGroupNamePattern='APIAPIFunction')
+    log_group_names = [group['logGroupName'] for group in response['logGroups']]
+    end_time = int(time.time()) * 1000
+    start_time = end_time - 1 * 24 * 60 * 60 * 1000  # recent 1 days logs
+
+    zip_bytes = io.BytesIO()
+    with zipfile.ZipFile(zip_bytes, 'w') as zipf:
+        for log_group_name in log_group_names:
+            response = logs.filter_log_events(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `filter_log_events` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)